### PR TITLE
Update function archetype quick start

### PR DESCRIPTION
**Pull request by Flanker32**
Update function archetype quick start, fixes https://msazure.visualstudio.com/Unified%20Platform%20KPIs/_workitems/edit/7494464

### CODE REVIEW

File: azure-functions-archetype/README.md [hunk 1]
```diff
@@ -3,39 +3,60 @@
...
+## Documentation
+Following document is based on this archetype
+- [Quickstart](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function-azure-cli?tabs=bash%2Cbrowser&pivots=programming-language-java): Create a function in Azure that responds to HTTP requests
 
-Like any other Maven Archetype, you are required to provide values for parameters `groupId` and `artifactId`, the `version` and `package` will have default values if not set.
+## Usage
+
+To generate an Azure Functions project, please adjust the following parameters to your needs.
+
+- Adjust `-DgroupId=com.ms` to define the base Maven group ID
```
Comment (andxu) [inline azure-functions-archetype/README.md:14]: replace it with a table for  configurable arguments

Comment (andxu) [reply azure-functions-archetype/README.md:14]: Property | Type | Required | Default Value | Description
-- | -- | -- | --

File: azure-functions-archetype/README.md [hunk 2]
```diff
@@ -3,39 +3,60 @@
...
 
-## Azure Parameters
+## Documentation
+Following document is based on this archetype
+- [Quickstart](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function-azure-cli?tabs=bash%2Cbrowser&pivots=programming-language-java): Create a function in Azure that responds to HTTP requests
 
-Like any other Maven Archetype, you are required to provide values for parameters `groupId` and `artifactId`, the `version` and `package` will have default values if not set.
+## Usage
+
+To generate an Azure Functions project, please adjust the following parameters to your needs.
```
Comment (andxu) [inline azure-functions-archetype/README.md:12]: adjust  -> specify

File: azure-functions-archetype/README.md [hunk 3]
```diff
@@ -3,39 +3,60 @@
 
 This is the Maven Archetype for Azure Functions.
 
-## Azure Parameters
+## Documentation
+Following document is based on this archetype
```
Comment (andxu) [inline azure-functions-archetype/README.md:7]: Please follow the this quickstart document for how to use this archetype

File: azure-functions-archetype/README.md [hunk 4]
```diff
@@ -3,39 +3,60 @@
...
+    -DartifactId=azure-function-demo \
+    -DappName=MyFunctionApp \
+    -DappRegion=westus \
+    -DappServicePlanName=java-functions-app-service-plan \
+    -DresourceGroup=java-functions-group \
+    -DjavaVersion=8
+```
+> To create a function project with docker support, please add `-Ddocker` to above command
+
+You may also remove `-B` and all `-D` parameters to set the values in interactive mode.
```
Comment (andxu) [inline azure-functions-archetype/README.md:36]: If you remove the `-B` argument, you will enter the interactive mode, all the parameters in above table will be promoted to you.

File: azure-functions-archetype/README.md [hunk 5]
```diff
@@ -3,39 +3,60 @@
...
+```bash
+mvn archetype:generate -DarchetypeGroupId=com.microsoft.azure -DarchetypeArtifactId=azure-functions-archetype
+```
 
-On top of that, five extra Azure parameters are required for Azure Functions deployment. Default value and description for those parameters are listed in below table, you can use their default values or change to your own value with command line options.
+## Available Parameters
 
 Parameter Name | Default Value | Description
 ---|---|---
+`groupId`| | Base Maven group ID
```
Comment (andxu) [inline azure-functions-archetype/README.md:46]: the group id

File: azure-functions-archetype/README.md [hunk 6]
```diff
@@ -3,39 +3,60 @@
...
+`artifactId`| | Base Maven artifact Id
+`package` | `${groupId}` | Java source package
```
Comment (andxu) [inline azure-functions-archetype/README.md:48]: I don't know from the word: "Java source package", can you describe it more clear?

File: azure-functions-archetype/README.md [hunk 7]
```diff
@@ -3,39 +3,60 @@
...
 `appName` | `${artifactId}-${timestamp}` | Specifies the name of your Azure Functions, which will be used to package, run and deploy your project.
```
Comment (andxu) [inline azure-functions-archetype/README.md:49]: Remove all Specifies

File: azure-functions-archetype/README.md (hunk 1 above)
Comment (Flanker32) [reply azure-functions-archetype/README.md:14]: Table was provided in the available parameters part

File: azure-functions-archetype/README.md (hunk 2 above)
Comment (Flanker32) [reply azure-functions-archetype/README.md:12]: As we provided example values here, so uses adjust

File: azure-functions-archetype/README.md [hunk 8]
```diff
@@ -3,39 +3,61 @@
...
+    -DappName=MyFunctionApp \
+    -DappRegion=westus \
+    -DappServicePlanName=java-functions-app-service-plan \
+    -DresourceGroup=java-functions-group \
+    -DjavaVersion=8
 ```
+> To create a function project with docker support, please add `-Ddocker` to above command
 
-To create a function project with docker support, please add -Ddocker to above command
+If you remove the -B argument, you will enter the interactive mode, all the parameters in above table will be promoted to you.
```
Comment (andxu) [inline azure-functions-archetype/README.md:37]: `-B`

Comment (andxu) [reply azure-functions-archetype/README.md:37]: in above table -> mentioned above

**Review by andxu — APPROVED** [review]

**Pull request merged**